main/lsp/requests/definition.cc

```diff
@@ -9,6 +11,26 @@ DefinitionTask::DefinitionTask(const LSPConfiguration &config, MessageId id,
                                unique_ptr<TextDocumentPositionParams> params)
     : LSPRequestTask(config, move(id), LSPMethod::TextDocumentDefinition), params(move(params)) {}
 
+core::Loc
+DefinitionTask::findRequireRelativeLoc(const core::GlobalState &gs,
+                                       const std::vector<std::unique_ptr<core::lsp::QueryResponse>> &responses) {
+    if (responses.size() < 2)
+        return core::Loc::none();
+
+    auto parentSendResp = move(responses[1])->isSend();
+    const bool isRequireRelative = parentSendResp && parentSendResp->callerSideName == core::Names::require_relative();
+    auto literal = move(responses[0])->isLiteral();
+    if (isRequireRelative && literal) {
+        auto literalValue = core::cast_type_nonnull<core::LiteralType>(literal->retType.type).asName(gs).shortName(gs);
+        auto relativeFileName = fmt::format("{}.rb", literalValue);
+        auto baseFilePath = std::filesystem::path(config.uri2FileRef(gs, params->textDocument->uri).data(gs).path());
+        auto targetFilePath = baseFilePath.replace_filename(relativeFileName);
+        auto loc = core::Loc(gs.findFileByPath(targetFilePath.string()), 0, 0);
```

Comment:
You're going to have to canonicalize the path so it contains the exact workspace root. This may be `~/stripe/foo`, or `.`, or `../foo/...`, or whatever. We don't canonicalize it AFAIK.

LSPConfiguration has a rootPath you can use for this:
https://github.com/sorbet/sorbet/blob/a9063fd1/main/lsp/LSPConfiguration.h#L83

I'd suggest:
* Resolving the `require_relative` path to an absolute path
* Replacing the absolute path of the workspace root with the specific `rootPath` form.
* Then finding the file with that canonicalized path.

Reply:
See `localName2Remote` for more information here; this shows you how to take a `core::File`'s path and turn it into a URI.

https://github.com/sorbet/sorbet/blob/a9063fd1/main/lsp/LSPConfiguration.cc#L122